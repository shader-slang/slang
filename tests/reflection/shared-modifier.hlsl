// shared-modifier.hlsl
//TEST:REFLECTION:-profile ps_5_0 -target hlsl

// Confirm that we expose the `shared` modifier in reflection data.

Texture2D t;
shared SamplerState s;

float4 main(float2 uv : UV) : SV_Target
{
	return t.Sample(s, uv);
}