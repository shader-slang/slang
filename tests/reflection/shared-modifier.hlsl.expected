result code = 0
standard error = {
}
standard output = {
{
    "parameters": [
        {
            "name": "t",
            "binding": {"kind": "shaderResource", "index": 0},
            "type": {
                "kind": "resource",
                "baseShape": "texture2D"
            }
        },
        {
            "name": "s",
            "shared": true,
            "binding": {"kind": "samplerState", "index": 0},
            "type": {
                "kind": "samplerState"
            }
        }
    ],
    "entryPoints": [
        {
            "name": "main",
            "stage:": "fragment",
            "parameters": [
                {
                    "name": "uv",
                    "stage": "fragment",
                    "binding": {"kind": "varyingInput", "index": 0},
                    "semanticName": "UV",
                    "type": {
                        "kind": "vector",
                        "elementCount": 2,
                        "elementType": {
                            "kind": "scalar",
                            "scalarType": "float32"
                        }
                    }
                }
            ],
            "result:": {
                "stage": "fragment",
                "binding": {"kind": "varyingOutput", "index": 0},
                "semanticName": "SV_TARGET",
                "type": {
                    "kind": "vector",
                    "elementCount": 4,
                    "elementType": {
                        "kind": "scalar",
                        "scalarType": "float32"
                    }
                }
            }
        }
    ]
}
}
